### PR TITLE
Embed LALRPOP version in generated files, and check it on rebuild

### PR DESCRIPTION
**Pull request by mchesser**
Addresses #243

### CODE REVIEW

**Comment by nikomatsakis** [comment]
Oh, hey, nice! Sorry I've been slow on the reviews! I've been traveling. (And indeed have some more travels coming up...)

File: lalrpop/src/build/mod.rs [hunk 1]
```diff
@@ -157,6 +165,14 @@ fn needs_rebuild(lalrpop_file: &Path, rs_file: &Path) -> io::Result<bool> {
                                   -> bool {
         true
     }
+
+    fn compare_lalrpop_version(rs_file: &Path) -> io::Result<bool> {
+        let mut input_str = String::new();
+        let mut f = io::BufReader::new(try!(fs::File::open(&rs_file)));
+        try!(f.read_line(&mut input_str));
+
+        Ok(input_str.trim() != LALRPOP_VERSION_HEADER)
```
Comment (nikomatsakis) [inline lalrpop/src/build/mod.rs]: Very nice.

I wonder if there is some way we can test this... well, we're not testing it now anyway I guess. We should think about it for the future though!

**Comment by nikomatsakis** [comment]
@mchesser I presume you've tested this locally and made sure it does indeed prevent "recompilation" most of the time? (I can't easily test myself just now...)

**Comment by mchesser** [comment]
Yeah, I tested it with the CLI and with a project that uses it as a build dependency and there didn't appear to be any issues.

**Comment by nikomatsakis** [comment]
Thanks! :)

**Pull request merged**